lib/HLSL/DxcOptimizer.cpp

```diff
@@ -47,6 +47,13 @@ using namespace hlsl;
 inline static bool wcseq(LPCWSTR a, LPCWSTR b) {
   return 0 == wcscmp(a, b);
 }
+inline static bool wcsstarts(LPCWSTR value, LPCWSTR prefix) {
+  while (*value && *prefix && *value == *prefix) {
```

Comment:
Kaboom. Much like other str* functions. I could add an assert too, but it'll AV on contact so it should be straightforward to blame caller from that.